uboot/uboot.env

```diff
@@ -8,58 +8,60 @@
 
 nerves_fw_active=a
 
-# nerves_fw_autovalidate controls whether updates are considered valid once
-# applied. If set to 0, the user needs to set nerves_fw_validated to 1 in their
-# application. If they don't set it before a reboot, then the previous software
-# is run. If 1, then no further action needs to be taken.
-nerves_fw_autovalidate=1
+# nerves_fw_autovalidate isn't supported in this system.
+# The user needs to explicitly validate the firmware in their application,
+# otherwise if the firmware isn't validated before a reboot, the previous
+# firmware will be set as active.
+# Other Nerves systems generally set `nerves_fw_autovalidate=0`, but this
+# step is skipped as the fallback has been removed.
 
 # nerves_fw_validated is 1 if the current boot selection is accepted It is set
 # to 1 here, since this environment is written in the factory, so it is
 # implicitly valid.
-nerves_fw_validated=1
+a.nerves_fw_validated=1
 
 # nerves_fw_booted is 0 for the first boot and 1 for all reboots after that.
 # NOTE: Keep this '0' so that all new boards run a 'saveenv' to exercise the
 #       code that writes back to the eMMC early on.
 nerves_fw_booted=0
 
+# For U-Boot's bootcount feature
+upgrade_available=0
+bootcount=0
+bootlimit=4
+
 # The nerves initialization logic
 #
 # The nerves_init code is run at boot (see the last line of the file). It
 # checks whether this is a first boot or not. If it's not the first boot, then
 # the firmware better be validated or it reverts to running the firmware on
 # the opposite partition.
-nerves_revert=\
-    if test ${nerves_fw_active} = "a"; then\
-        echo "Reverting to partition B";\
-        setenv nerves_fw_active "b";\
-    else\
-        echo "Reverting to partition A";\
-        setenv nerves_fw_active "a";\
-    fi
-
-nerves_init=\
-    if test ${nerves_fw_booted} = 1; then\
-        if test ${nerves_fw_validated} = 0; then\
-            run nerves_revert;\
-            setenv nerves_fw_validated 1;\
-            saveenv;\
-        fi;\
-    else\
-        setenv nerves_fw_booted 1;\
-        if test ${nerves_fw_autovalidate} = 1; then\
-            setenv nerves_fw_validated 1;\
-        fi;\
-        saveenv;\
-    fi;\
-    if test ${nerves_fw_active} = "a"; then\
-        setenv uenv_root /dev/mmcblk2p1;\
-        setenv bootpart 2:1;\
-    else\
-        setenv uenv_root /dev/mmcblk2p2;\
-        setenv bootpart 2:2;\
-    fi
+nerves_revert= \
+    if test ${nerves_fw_active} = "a"; then \
+        echo "Reverting to partition B"; \
+        setenv nerves_fw_active "b"; \
+        setenv b.nerves_fw_validated "1"; \
```

Comment:
I think we should trust the app to do the validation. There's a Nerves mechanism that prevents reverting to unvalidated firmware, which we don't want to accidentally bypass. I haven't tried it, but I feel like calling revert twice in a row with this code may be possible to revert to unvalidated firmware, which will then be marked valid.